_posts/2023-02-13-Branch-Review-and-Learn.md

```diff
@@ -0,0 +1,135 @@
+
+## Concept: Coursework Branching
+
+<pre class="mermaid">
+flowchart LR
+ subgraph Course
+    direction LR
+    open["open\nassignment"]
+    turn_in["review\nassignment"]
+  end
+  subgraph Learner ["&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Learner"]
+    direction LR
+    work["completed\nassignment"]
+  end
+  open -.-> turn_in
+  open --> |works towards| work
+  work --> |seeks review| turn_in
+</pre>
+<script type="module">
+  import mermaid from 'https://unpkg.com/mermaid@9/dist/mermaid.esm.min.mjs';
+  mermaid.initialize({ startOnLoad: true });
+</script>
+
+_An example course and learner assignment workflow._
+{:.center}
+
+[Git branching](https://www.atlassian.com/git/tutorials/using-branches) practices may be understood in context with similar workflows from real life. Consider a scenario within an educational course where an assignment is made available to a learner. In addition to the steps shown in the diagram above, it's important to think about _why_ this pattern is beneficial:
+
+- Completing an assignment allows us as social, inter-dependent beings to present new findings which enable learning and amalgamation of additional ideas from others.
+- The timebound nature of assignments enables us to practice some form of [timeboxing](https://cu-dbmi.github.io/set-website/2023/01/17/Timebox-Your-Software-Work.html) so as to minimize tasks which may take too much time.
+- Segmenting applied learning in distinct, goal-orientated chunks helps make larger topics easier to understand.
+
+## Branching to Complete an "Assignment"
+
+<pre class="mermaid">
+%%{init: { 'logLevel': 'debug', 'theme': 'default' , 'themeVariables': {
+      'git0': '#4F46E5',
+      'git1': '#10B981',
+      'gitBranchLabel1': '#ffffff'
+} } }%%
+    gitGraph
+       commit id: "..."
+       commit id: "opened"
+       branch learner
+       checkout learner
+       commit id: "completed"
+       checkout main
+</pre>
+
+_An example git diagram showing learner branch based off main._
+{:.center}
+
+Following the course assignment workflow, the diagram above shows an in-progress `learner` branch based off of the `main` branch. When the `learner` branch is created, we bring into it everything we know from `main` (the course) so far in the form of [commits](https://github.com/git-guides/git-commit), or groups of changes to various files. [Branching](https://github.com/git-guides#create-a-branch) allows us to make consistent and well described changes based on what's already happened without impacting others work in the meantime.
```

Comment:
First, I love these Git diagrams! I like the example as well, but I think naming the branch `learner` gets away from the idea you bring up elsewhere of using branches for small tasks. Perhaps naming it `assignment` or some other indication that it's a task, not the student's personal branch, would make things clearer.

Reply:
Love this, thanks! It definitely aligns well to the purpose rather than role for that lane/branch to demonstrate the significance here. I'll make changes accordingly.